components/brave_wallet/common/bitcoin_utils_unittest.cc

```diff
@@ -0,0 +1,145 @@
+
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/common/bitcoin_utils.h"
+
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_util.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+TEST(BitcoinUtilsUnitTest, Bip0173TestVectors1) {
+  // https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#examples
+
+  std::vector<uint8_t> pubkey;
+  ASSERT_TRUE(base::HexStringToBytes(
+      "0279BE667EF9DCBBAC55A06295CE870B07029BFCDB2DCE28D959F2815B16F81798",
+      &pubkey));
+
+  EXPECT_EQ(PubkeyToSegwitAddress(pubkey, true),
+            "tb1qw508d6qejxtdg4y5r3zarvary0c5xw7kxpjzsx");
+  EXPECT_EQ(PubkeyToSegwitAddress(pubkey, false),
+            "bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t4");
+
+  // TODO(apaymyshev): support P2WSH.
+}
+
+TEST(BitcoinUtilsUnitTest, Bip0173TestVectors2) {
+  // https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#test-vectors
+
+  struct ValidTestCase {
+    const char* address;
+    const char* opcodes;  // not relevant parts of scriptpubkey.
```

Comment:
unused?

Reply:
This is a midle part of each test_case which is not needed by this test, but exists in test vectors bip 173. Decided to keep it for easy comparison with test vectors